core/src/test/scala/unit/kafka/utils/TestUtils.scala

```diff
@@ -1865,4 +1866,17 @@ object TestUtils extends Logging {
       timedOut.set(true)
     }
   }
+
+  /**
+   * Resets the logging configuration after the test.
+   */
+  def resetLogging[T] = {
+    org.apache.log4j.LogManager.resetConfiguration()
+    val stream = this.getClass.getResourceAsStream("/log4j.properties")
+    try {
+      PropertyConfigurator.configure(stream)
+    } finally {
+      stream.close()
+    }
```

Comment:
I wonder we should fix the issue with this workaround at this moment since we are upgrading to log4j2 which providing a convenient way to reload configuration.

https://logging.apache.org/log4j/2.x/javadoc/log4j-core/org/apache/logging/log4j/core/config/Configurator.html#reconfigure()

Reply:
Thanks for the input, @frankvicky. You're absolutely right—we should use the method you suggested once we upgrade to log4j2. I appreciate you pointing that out. If the log4j upgrade PR gets merged into trunk first, I'll make sure to rebase my PR. Alternatively, if this PR is merged beforehand, I’d be happy to help replace all instances of PropertyConfigurator.configure with Configurator.reconfigure, if needed. (though I think the log4j2 upgrade part should have been done all in once, but I can help submit patch for you if needed)

Reply:
So this PR is depends on [upgrading to Log4j2 PR](https://github.com/apache/kafka/pull/17373), right?

Reply:
It's related, we need to modify `PropertyConfigurator.configure(stream)` after https://github.com/apache/kafka/pull/17373 merged.

Reply:
Got it! Thanks.

Reply:
We can merge this small PR first. #17373 can address both the failures and conflicts later.